File: python/Neetcode/TwoPointers/Easy/1984.MinimumDifferenceKScores.py
# Pick the scores of any k students from the array so that the difference 
# between the highest and the lowest of the k scores is minimized.

def minimumDifference(self, nums, k):
    nums.sort()

    left, right = 0, k - 1
    result = float('inf')

    while right < len(nums):
        result = min(result, nums[right] - nums[left])

        left, right = left + 1, right + 1

    return result

print(minimumDifference([90], 1))
print(minimumDifference([9,4,1,7], 2))

File: swift/Neetcode/TwoPointers/Easy/1984.MinimumDifferenceKScores.swift
//Pick the scores of any k students from the array so that the difference 
//between the highest and the lowest of the k scores is minimized.

func minimumDifference(_ nums: [Int], _ k: Int) -> Int {
    var nums = nums.sorted()
    var result = Int.max
    var left = 0

    for i in (k - 1)..<nums.count {
        result = min(result, nums[i] - nums[left])
        left += 1
    }

    return result
}

print(minimumDifference([90], 1))
print(minimumDifference([9,4,1,7], 2))